Plugins/Statistics/GitImpact/ImpactControl.cs

```diff
@@ -393,71 +394,84 @@ private void UpdatePathsAndLabels()
                 _lineLabels.Clear();
 
                 // Add points to each author's GraphicsPath
-                foreach (var author_points in author_points_dict)
+                foreach (var (author, points) in author_points_dict)
                 {
-                    string author = author_points.Key;
-
                     // Scale heights
-                    for (int i = 0; i < author_points.Value.Count; i++)
+                    for (int i = 0; i < points.Count; i++)
                     {
-                        author_points.Value[i] = Tuple.Create(
-                            new Rectangle(author_points.Value[i].Item1.Left, (int)(author_points.Value[i].Item1.Top * height_factor),
-                                          author_points.Value[i].Item1.Width, Math.Max(1, (int)(author_points.Value[i].Item1.Height * height_factor))),
-                            author_points.Value[i].Item2);
+                        var (unscaledRect, num) = points[i];
+
+                        var rect = new Rectangle(unscaledRect.Left, (int)(unscaledRect.Top * height_factor),
+                            unscaledRect.Width, Math.Max(1, (int)(unscaledRect.Height * height_factor)));
+
+                        points[i] = Tuple.Create(rect, num);
```

Comment:
`points` elements are reassigned here, which becomes more obvious when there aren't so many array accesses everywhere.

Reply:
I am still not convinced this is the same or that this code works at all...

Reply:
One good thing with rendering algorithms like this is that bugs usually manifest very visually.